pywinauto/controls/uia_controls.py

```diff
@@ -575,17 +605,33 @@ def item_count(self):
             # TODO: This could be implemented by getting custom ItemCount Property using RegisterProperty
             # TODO: See https://msdn.microsoft.com/ru-ru/library/windows/desktop/ff486373%28v=vs.85%29.aspx for details
             # TODO: comtypes doesn't seem to support IUIAutomationRegistrar interface
-            return (len(self.children()))
+            return self.__resolve_row_count(len(self.children()))
 
     # -----------------------------------------------------------
     def column_count(self):
         """Return the number of columns"""
         if self.iface_grid_support:
             return self.iface_grid.CurrentColumnCount
+        elif self.is_table and len(self.children()) > 0:
+            return self.__resolve_col_count(len(self.children()[0].children()))
```

Comment:
Method `.children()` returns all children which might be not optimal. Getting first child only is enough to check its count is not zero. This is relevant for all `.children()[0]` calls.